linera-chain/src/manager/multi.rs

```diff
@@ -307,12 +308,17 @@ impl MultiOwnerManager {
         key_pair: Option<&KeyPair>,
         now: Timestamp,
     ) {
+        let round = certificate.round;
+        // Validators only change their locked block if the new one is included in a proposal in the
+        // current round, or it is itself in the current round.
+        if key_pair.is_some() && round < self.current_round() {
+            return;
+        }
         let Some(value) = certificate.value.clone().into_confirmed() else {
             // Unreachable: This is only called with validated blocks.
             error!("Unexpected certificate; expected ValidatedBlock");
             return;
         };
-        let round = certificate.round;
         self.update_timeout(round, now);
```

Comment:
Does update_timeout make sense if we're not a validator? (just checking)

Reply:
Yes, updating the locked block increases the current round to at least that block's round, both for validators and for clients.

But I just realized that the locked block also needs to be taken into account in `next_block`. Will add a fix…